nixos/modules/installer/tools/nixos-prepare-root.sh

```diff
@@ -0,0 +1,120 @@
+#! @shell@
+
+# This script's goal is to perform all "static" setup of a filesystem structure from pre-built store paths. Everything
+# in here should run in a non-root context and inside a Nix builder. It's designed primarily to be called from image-
+# building scripts and from nixos-install, but because it makes very few assumptions about the context in which it runs,
+# it could be useful in other contexts as well.
+#
+# Current behavior:
+#  - set up basic filesystem structure
+#  - make Nix store etc.
+#  - copy Nix, system, channel, and misceallaneous closures to target Nix store
+#  - register validity of all paths in the target store
+#  - set up channel and system profiles
+
+# Ensure a consistent umask.
+umask 0022
+
+set -e
+
+mountPoint="$1"
```

Comment:
I'm not going to pretend this has a user-friendly CLI. It's mostly intended to be used by `nixos-install.sh` and the upcoming new image building machinery right now. If someone has a suggestion on how to remove the duplicate closure vs. path stuff in this CLI, I'm all ears.